Reportedly/Modules/LoginEmail/LoginEmailViewController.swift

```diff
@@ -0,0 +1,285 @@
+        
+        blurEffectView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+        
+        titleContainerView.snp.makeConstraints {
+            $0.centerX.leading.equalToSuperview()
+            $0.top.equalTo(view.snp.topMargin)
+            $0.bottom.equalTo(containerStackView.snp.top)
+        }
+        
+        titleLabel.snp.makeConstraints {
+            $0.centerX.centerY.leading.equalToSuperview()
+        }
+        
+        containerStackView.snp.makeConstraints {
+            $0.centerX.centerY.equalToSuperview()
+            $0.width.equalToSuperview().multipliedBy(0.85)
+        }
+        
+        emailField.snp.makeConstraints {
+            $0.height.equalTo(56.0)
+        }
+        
+        passwordField.snp.makeConstraints {
+            $0.height.equalTo(emailField.snp.height)
+        }
+        
+        loginButton.snp.makeConstraints {
+            $0.height.equalTo(56.0)
+        }
+        
+        signupLabelWithLinkView.snp.makeConstraints {
+            $0.leading.centerX.equalToSuperview()
+            $0.bottom.equalTo(view.safeAreaInsets.bottom).inset(24)
+        }
+    }
+    
+    private func setUpViews() {
+        setUpSuperview()
+        setUpBackgroundImageView()
+        setUpTitleLabel()
+        setUpContainerStackView()
+        setUpEmailField()
+        setUpPasswordField()
+        setUpLoginButton()
+        setUpSignupLabelWithLinkView()
+    }
+    
+    private func setUpSuperview() {
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard(_:)))
+        view.addGestureRecognizer(tap)
+    }
+    
+    private func setUpBackgroundImageView() {
+        backgroundImageView.contentMode = .scaleAspectFill
+        backgroundImageView.image = Asset.bg_launch_screen()
+    }
+    
+    private func setUpTitleLabel() {
+        titleLabel.font = UIFont.boldSystemFont(ofSize: 38)
+        titleLabel.numberOfLines = 1
+        titleLabel.textAlignment = .center
+    }
+    
+    private func setUpContainerStackView() {
+        containerStackView.axis = .vertical
+        containerStackView.spacing = 20
+    }
+    
+    private func setUpEmailField() {
+        emailField.keyboardType = .emailAddress
+        emailField.returnKeyType = .next
+        emailField.addHorizontalPadding(withValue: 8)
+        emailField.addTarget(self, action: #selector(textFieldDidChange), for: .editingChanged)
+        emailField.delegate = self
+    }
+    
+    private func setUpPasswordField() {
+        passwordField.isSecureTextEntry = true
+        passwordField.returnKeyType = .go
+        passwordField.addHorizontalPadding(withValue: 8)
+        passwordField.addTarget(self, action: #selector(textFieldDidChange), for: .editingChanged)
+        passwordField.delegate = self
+    }
+    
+    private func setUpLoginButton() {
+        loginButton.isEnabled = false
+        loginButton.titleLabel?.font = UIFont.boldSystemFont(ofSize: 16)
```

Comment:
Can we copy spacing constants from the1 to update these sizes? 😅

Reply:
yeah sure 🙏